cellmlmanip/model.py

```diff
@@ -410,27 +410,44 @@ def get_equation_graph(self, refresh=False) -> nx.DiGraph:
         self.graph = graph
         return graph
 
-    def get_equations_for(self, symbols, lexicographical_sort=True):
+    def get_equations_for(self, symbols, sort_by_input_symbols=False, excluded_symbols=[]):
         """Get all equations for given collection of symbols
 
         Results are sorted in topographical order.
         :param symbols: the symbols to get the equations for
-        :param lexicographical_sort: indicates whether the result is sorted in lexicographical order first
+        :param sort_by_input_symbols: indicates whether the sorting in symbols determines the sorting of the output
+        :param excluded_symbols: list of symbols to exlcude
         """
         graph = self.get_equation_graph()
-        if lexicographical_sort:
-            sorted_symbols = nx.lexicographical_topological_sort(graph, key=str)
-        else:
-            sorted_symbols = nx.topological_sort(graph)
 
-        # Create set of symbols for which we require equations
-        required_symbols = set()
+        def get_parents(node, graph=graph):
```

Comment:
would this function not be useful as a global function ?

Reply:
Perhaps I didn't  really look at the whole library. If you think it's useful feel free to put it elsewhere. This pull request was just me experimenting.

The main thing I'm wondering is whether it's correct. Are there ever any case where direct ancestors have an interdependence (e.g. x=5y, y=5)? (If so order is important if I'm able to generate C++ code)

Reply:
Not sure what you mean by direct ancestors having an interdependence?

Reply:
ok so lets say I have `x = 5*y + 6*z`
ancestors:
y and z
 would it be possible for y and z then t have their own equations for example:
`y =2*z
z = 1`

Reply:
Almost certainly yes!

Reply:
So we somehow need to make sure z ends up before y in the example somehow

Reply:
Would a topological sort work?

Reply:
I thought get_equations_for was doing a topological sort? Using the nx graph library? Certainly that's the correct approach.

The trickier bit is determining a consistent relative ordering where there isn't a direct dependency.

Reply:
It does but not with my flag. i've changed that now